travelwise/api/views.py

```diff
@@ -119,6 +147,31 @@ def delete_activity(function_arguments):
         
         return f'Deleted activity with id \"{id}\"'
         
+    def update_activity(function_arguments):
+        id = function_arguments.get('id')
+        start_time = function_arguments.get('start_time')
+        end_time = function_arguments.get('end_time')
+        note = function_arguments.get('note')
+        
+        message = f'Updated activity with id \"{id}\"'
+        
+        activity = Activity.objects.filter(pk=id, plan=travel_plan).first()
+        if activity is None:
+            return f'Activity with id \"{id}\" does not exist'
+        
+        if start_time is not None:
+            activity.start_time = start_time
+            message += f', Start Time set to {start_time}'
+        if end_time is not None:
+            activity.end_time = end_time
+            message += f', End Time set to {end_time}'
```

Comment:
From my testing, the AI is smart enough to understand that it needs to update both when asked. I was also trying to consider the scenario where the user asks for an activity to take 3 hours vs 1 for example.

Reply:
Do you prefer breaking it up into to two guards? Or should we put it into one?

Reply:
The logging here is mostly cosmetic. 2 guards are needed in case the AI tries to set one but leaves the other null, which will make the function attempt to null that field. If we introduce a new guard, it would be in addition to the other 2.